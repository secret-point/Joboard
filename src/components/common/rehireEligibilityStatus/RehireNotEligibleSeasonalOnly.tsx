import React from "react";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import { translate as t } from "../../../utils/translator";
import { Job } from "../../../utils/types/common";

interface RehireNotEligibleSeasonalOnlyProps {
  jobDetail?: Job
}

const RehireNotEligibleSeasonalOnly = (props: RehireNotEligibleSeasonalOnlyProps) => {

  const { jobDetail } = props;

  const titleWithJobTitle = t("BB-rehire-not-eligible-seasonal-only-title-text-with-jobTitle", "Thank you for your interest in the {jobTitle} position. Our records indicate that you previously worked at Amazon.  At this time, you are eligible for rehire as a Seasonal Amazon Associate or third-party agency employee based upon your employment record. As a result, we are unable to move forward with your candidacy for this job. If you have any questions regarding rehire eligibility, contact the site HR Team.", {jobTitle: jobDetail?.jobTitle || ""});
  const titleWithoutJobTitle = t("BB-rehire-not-eligible-seasonal-only-title-text-without-jobTitle", "Thank you for your interest in this position. Our records indicate that you previously worked at Amazon.  At this time, you are eligible for rehire as a Seasonal Amazon Associate or third-party agency employee based upon your employment record. As a result, we are unable to move forward with your candidacy for this job. If you have any questions regarding rehire eligibility, contact the site HR Team.");

  return (
    <Col gridGap="S300">
      <Text>
        {
          jobDetail?.jobTitle ? titleWithJobTitle : titleWithoutJobTitle
        }
      </Text>
      <Text>
        {t("BB-rehire-not-eligible-seasonal-only-content-text", "Please visit our Seasonal Page to explore seasonal opportunities in your area.")}
      </Text>
    </Col>
  )
}

export default RehireNotEligibleSeasonalOnly;
