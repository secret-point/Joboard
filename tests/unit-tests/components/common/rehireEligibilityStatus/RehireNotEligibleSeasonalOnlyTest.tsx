import React from "react";
import { shallow } from "enzyme";
import RehireNotEligibleSeasonalOnly from "../../../../../src/components/common/rehireEligibilityStatus/RehireNotEligibleSeasonalOnly";
import { TEST_JOB2 } from "../../../../test-utils/test-data";


describe("RehireNotEligibleSeasonalOnly", () => {
    it("should match snapshot with jobDetail", () => {
        const shallowWrapper = shallow(<RehireNotEligibleSeasonalOnly jobDetail={ TEST_JOB2 }/>);
    
        expect(shallowWrapper).toMatchSnapshot();
    });

    it("should match snapshot without jobDetail", () => {
        const shallowWrapper = shallow(<RehireNotEligibleSeasonalOnly/>);
    
        expect(shallowWrapper).toMatchSnapshot();
    });
});