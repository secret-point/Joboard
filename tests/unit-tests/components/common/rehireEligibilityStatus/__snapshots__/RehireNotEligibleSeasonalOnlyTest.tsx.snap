// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RehireNotEligibleSeasonalOnly should match snapshot with jobDetail 1`] = `
<Col
  gridGap="S300"
>
  <Text
    font="primary"
    fontSize="T300"
    hyphens="none"
  >
    Thank you for your interest in the {jobTitle} position. Our records indicate that you previously worked at Amazon.  At this time, you are eligible for rehire as a Seasonal Amazon Associate or third-party agency employee based upon your employment record. As a result, we are unable to move forward with your candidacy for this job. If you have any questions regarding rehire eligibility, contact the site HR Team.
  </Text>
  <Text
    font="primary"
    fontSize="T300"
    hyphens="none"
  >
    Please visit our Seasonal Page to explore seasonal opportunities in your area.
  </Text>
</Col>
`;

exports[`RehireNotEligibleSeasonalOnly should match snapshot without jobDetail 1`] = `
<Col
  gridGap="S300"
>
  <Text
    font="primary"
    fontSize="T300"
    hyphens="none"
  >
    Thank you for your interest in this position. Our records indicate that you previously worked at Amazon.  At this time, you are eligible for rehire as a Seasonal Amazon Associate or third-party agency employee based upon your employment record. As a result, we are unable to move forward with your candidacy for this job. If you have any questions regarding rehire eligibility, contact the site HR Team.
  </Text>
  <Text
    font="primary"
    fontSize="T300"
    hyphens="none"
  >
    Please visit our Seasonal Page to explore seasonal opportunities in your area.
  </Text>
</Col>
`;
